ksmt-cvc5/src/main/kotlin/org.ksmt.solver.cvc5/KCvc5Context.kt

```diff
@@ -0,0 +1,158 @@
+package org.ksmt.solver.cvc5
+
+import io.github.cvc5.Solver
+import io.github.cvc5.Sort
+import io.github.cvc5.Term
+import org.ksmt.decl.KDecl
+import org.ksmt.expr.KExpr
+import org.ksmt.sort.KSort
+import org.ksmt.sort.KUninterpretedSort
+import java.util.TreeMap
+
+class KCvc5Context(private val solver: Solver) : AutoCloseable {
+    private var isClosed = false
+
+    // we can't use HashMap with Term and Sort (hashcode is not implemented)
+    private val expressions = HashMap<KExpr<*>, Term>()
+    private val cvc5Expressions = TreeMap<Term, KExpr<*>>()
+    private val sorts = HashMap<KSort, Sort>()
+    private val cvc5Sorts = TreeMap<Sort, KSort>()
+    private val decls = HashMap<KDecl<*>, Term>()
+    private val cvc5Decls = TreeMap<Term, KDecl<*>>()
+
+    internal val uninterpretedSorts = hashSetOf<KUninterpretedSort>()
+    internal val declarations: Set<KDecl<*>>
```

Comment:
It is not correct due to the `pop` operation.

Reply:
We can collect uninterpreted sorts and declarations like tracked vars, based on push-levels (last commit).